src/components/Drawer.js

```diff
@@ -0,0 +1,146 @@
+const React = require('react');
+const Velocity = require('velocity-animate');
+
+const StyleConstants = require('../constants/Style');
+
+const Icon = require('../components/Icon');
+
+const Drawer = React.createClass({
+  propTypes: {
+    duration: React.PropTypes.number,
+    easing: React.PropTypes.array,
+    isOpen: React.PropTypes.bool,
```

Comment:
@cerinman I actually agree with your first statement. I think this component should not control it's open/close state. Instead the component should animate in on mount and animate out on unmount. The parent component should control when it's added to the UI.

@paniclater thoughts?